packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/index.ts

```diff
@@ -442,8 +459,107 @@ const analyticsMiddleware = () => (store) => (next) => (action) => {
           originalTimestamp: getOriginalTimestamp(),
           output_amount: outputAmount,
           output_currency: outputCurrency,
-          output_type: outputType,
-          type: AnalyticsType.EVENT
+          output_type: outputType
+        })
+
+        break
+      }
+
+      case AT.components.request.GET_NEXT_ADDRESS: {
+        const state = store.getState()
+        const accountType =
+          state.forms.requestCrypto.values.selectedAccount.type === 'CUSTODIAL'
+            ? 'TRADING'
+            : 'USERKEY'
```

Comment:
ideally all of this we should move to enums/const

Reply:
You're right! I'm converting these to enums